frontend/apps/reader/modules/readersearch.lua

```diff
@@ -51,33 +83,103 @@ function ReaderSearch:onShowFulltextSearchInput()
                     text = backward_text,
                     callback = function()
                         if self.input_dialog:getInputText() == "" then return end
-                        UIManager:close(self.input_dialog)
-                        self:onShowSearchDialog(self.input_dialog:getInputText(), 1)
+                        self.last_search_text = self.input_dialog:getInputText()
+                        self.use_regex = self.check_button_regex.checked
+                        self.case_insensitive = not self.check_button_case.checked
+                        if self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText()) ~= 0 then
+                            UIManager:show(InfoMessage:new{ text = _("Error in regular expression!") })
+                        else
+                            UIManager:close(self.input_dialog)
+                            self:onShowSearchDialog(self.input_dialog:getInputText(), 1, self.use_regex, self.case_insensitive)
+                        end
                     end,
                 },
                 {
                     text = forward_text,
                     is_enter_default = true,
                     callback = function()
                         if self.input_dialog:getInputText() == "" then return end
-                        UIManager:close(self.input_dialog)
-                        self:onShowSearchDialog(self.input_dialog:getInputText(), 0)
+                        self.last_search_text = self.input_dialog:getInputText()
+                        self.use_regex = self.check_button_regex.checked
+                        self.case_insensitive = not self.check_button_case.checked
+                        if self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText()) ~= 0 then
+                            UIManager:show(InfoMessage:new{ text = _("Error in regular expression!") })
+                        else
+                            UIManager:close(self.input_dialog)
+                            self:onShowSearchDialog(self.input_dialog:getInputText(), 0, self.use_regex, self.case_insensitive)
+                        end
                     end,
                 },
             },
         },
     }
+
+   -- checkboxes
+    self.check_button_regex = CheckButton:new{
+        text = _("Regular expression"),
+        face = Font:getFace("smallinfofont"),
```

Comment:
@hius07 thought this should be set generically in CheckButton.
There are not much users of CheckButton: frontend/ui/widget/frontlightwidget.lua frontend/ui/widget/openwithdialog.lua frontend/ui/widget/inputtext.lua
so it doesn't feel too risky.
If you'd rather keep that for @hius07 to do a PR, may be just don't include `face =` here, so there's no need to touch this file later.

Reply:
`smallinfofont` really looks better. I let this PR to @hius07